dev_docs/contributing/kibana_http_api_tf_guidelines.mdx

```diff
@@ -252,12 +252,24 @@ The cost of not catching errors early is interrupted workflow, repetitive plan/a
 
 ### Return as much as you can, and handle defaults carefully
 
-Terraform needs complete information to track resource state effectively. If your API doesn't return enough details, Terraform can't properly detect changes or manage drift.
+Terraform needs complete information to track resource state effectively. If your API doesn't return enough details, Terraform can't properly detect changes or manage drift. For this reason, **the overall recommendation is to _return as much information as possible and define the optional attributes in the [Terraform provider](https://github.com/elastic/terraform-provider-elasticstack/tree/main/internal/kibana) as `Optional: true, Default: <default value>`._**
 
 **The challenge with defaults**
 
 Many Kibana APIs have extensive configuration options. Requiring users to specify every single value would create verbose, unwieldy configurations. We need to keep in mind that changing a default may cause Terraform to get out-of-sync and so view changes to defaults as a breaking change until we can update our provider.
 
+[Our tests](https://github.com/elastic/kibana-team/issues/2001) show the following facts:
+
+* The API can return fields that Terraform doesn't understand (e.g.: a new version of Kibana returns extra fields in the response, or it returns fields that Terraform users shouldn't be concerned about like `last_updated` or `created_by`): Terraform dismisses them when comparing the resource.
+* Resource attributes defined as `Optional: true, Computed: true`: Terraform won't detect changes. This means we can change the defaults, choose not to return it, and users can manually change that attribute without Terraform noticing the change.
+* Defining a resource attribute as `Optional: true, Default: <default value>` provides a more consistent behavior to the user (it identifies external changes), but can tighten the Terraform provider's version support matrix: if Kibana changes the default value, Terraform won't know about the change until the Terraform resource configuration gets updated. Keep in mind that our Terraform provider doesn't follow the Stack release cycle, so we can't guarantee that the default value will be the same as the one in the Stack.
+
+This is why the overall recommendation is to return as much information as possible and define the optional attributes in the [Terraform provider](https://github.com/elastic/terraform-provider-elasticstack/tree/main/internal/kibana) as `Optional: true, Default: <default value>`.
+
+**So can't I return slimmed-down information?**
```

Comment:
FWIW I thought we could copy ES's `filter_path`, implementation:

https://github.com/elastic/kibana/pull/217332

But overall agree with this advice 👌🏻

Reply:
I added a note to the `filter_path` recommendation in https://github.com/elastic/kibana/pull/235412/commits/da27bb2d329deb64e910556805cf59edb69e9f18.